### PR TITLE
feat: generate database mocker operation name from sql parsing

### PR DESCRIPTION
**Pull request by yanni97**
support: SQL parses tableName

### CODE REVIEW

File: arex-storage-config/src/main/java/com/arextest/config/utils/parse/sqlparse/SqlParseManager.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
+package com.arextest.config.utils.parse.sqlparse;
```
Comment (coryhh) [inline arex-storage-config/src/main/java/com/arextest/config/utils/parse/sqlparse/SqlParseManager.java:1]: It is recommended to adjust the location of the parsing tool class. arex-storage-config is the management module for tables related to the configuration module and is referenced by arex-api.

File: arex-storage-web-api/src/main/java/com/arextest/storage/utils/DatabaseUtils.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+/**
+ * @author niyan
+ * @date 2024/4/24
+ * @since 1.0.0
+ */
+public class DatabaseUtils {
+
+    /**
+     * 新数据的operationName字段组成为：dbName-tableName,tableName,tableName-action-operationName; 旧数据的operationName字段组成为：operationName
```
Comment (pangdayuan1) [inline arex-storage-web-api/src/main/java/com/arextest/storage/utils/DatabaseUtils.java:13]: It is recommended to use English comments

File: arex-storage-web-api/src/main/java/com/arextest/storage/service/mockerhandlers/DatabaseMockerHandel.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+
+    public <T extends Mocker> T parseMocker(T item) {
+
+        String originOperationName = item.getOperationName();
+        Mocker.Target targetRequest = item.getTargetRequest();
+        String dbName = "";
+        String sqlBody = "";
+        if (targetRequest != null) {
+            dbName = (String) targetRequest.getAttribute(MockAttributeNames.DB_NAME);
```
Comment (pangdayuan1) [inline arex-storage-web-api/src/main/java/com/arextest/storage/service/mockerhandlers/DatabaseMockerHandel.java:52]: It is recommended to use the attributeAsString API

File: arex-storage-web-api/src/main/java/com/arextest/storage/service/mockerhandlers/DatabaseMockerHandel.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        }
+        // sqlParse
+        if (StringUtils.isNotBlank(sqlBody)) {
+            String[] splitSql = sqlBody.split(";");
+            StringBuilder basicInfo = new StringBuilder();
+            Map<String, String> tableAndAction = null;
+            for (String subSql : splitSql) {
+                tableAndAction = SqlParseManager.getInstance().parseTableAndAction(subSql);
+                if (MapUtils.isNotEmpty(tableAndAction)) {
+                    String action = tableAndAction.getOrDefault("action", "");
```
Comment (pangdayuan1) [inline arex-storage-web-api/src/main/java/com/arextest/storage/service/mockerhandlers/DatabaseMockerHandel.java:63]: use DbParseConstants.ACTION

File: arex-storage-config/src/main/java/com/arextest/config/utils/parse/sqlparse/SqlParseManager.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+import java.util.Map;
+
+/**
+ * @author niyan
+ * @date 2024/4/23
+ * @since 1.0.0
+ */
+@Slf4j
+public class SqlParseManager {
+    private static SqlParseManager INSTANCE;
```
Comment (binyu1005) [inline arex-storage-config/src/main/java/com/arextest/config/utils/parse/sqlparse/SqlParseManager.java:27]: 这里是否可以用spring的@component进行单例的管理？

File: arex-storage-web-api/src/main/java/com/arextest/storage/mock/impl/DefaultMockResultProviderImpl.java [hunk 6]
```diff
@@ -93,6 +94,11 @@ public <T extends Mocker> boolean putRecordResult(MockCategoryType category, Str
     // Obtain the number of the same interfaces in recorded data
     while (valueIterator.hasNext()) {
       T value = valueIterator.next();
+      // 新老数据兼容，上线一段时间后删除
```
Comment (binyu1005) [inline arex-storage-web-api/src/main/java/com/arextest/storage/mock/impl/DefaultMockResultProviderImpl.java:97]: It is recommended to use English comments

File: arex-storage-web-api/src/main/java/com/arextest/storage/mock/internal/matchkey/impl/DatabaseMatchKeyBuilderImpl.java [hunk 7]
```diff
@@ -168,7 +170,10 @@ private List<byte[]> dbMockKeyBuild(Mocker databaseMocker) {
     byte[] dbNameBytes = CacheKeyUtils.toUtf8Bytes(dbName);
     byte[] sqlTextBytes = CacheKeyUtils.toUtf8Bytes(sqlText);
     byte[] sqlParameterBytes = CacheKeyUtils.toUtf8Bytes(sqlParameter);
-    byte[] operationBytes = CacheKeyUtils.toUtf8Bytes(databaseMocker.getOperationName());
+
+    // 新老兼容逻辑，上线4天后可删除
```
Comment (binyu1005) [inline arex-storage-web-api/src/main/java/com/arextest/storage/mock/internal/matchkey/impl/DatabaseMatchKeyBuilderImpl.java:174]: It is recommended to use English comments

File: arex-storage-web-api/src/main/java/com/arextest/storage/service/mockerhandlers/DatabaseMockerHandel.java [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+                    String tableName = tableAndAction.getOrDefault("table", "");
+                    basicInfo.append(dbName).append('-').append(tableName).append('-').append(action).append("-").append(originOperationName).append(";");
+                }
+            }
+            item.setOperationName(basicInfo.toString());
```
Comment (coryhh) [inline arex-storage-web-api/src/main/java/com/arextest/storage/service/mockerhandlers/DatabaseMockerHandel.java:68]: Possible impact scenarios: Modifying operaname will cause the ignored compare type to not take effect.

File: arex-storage-web-api/src/main/java/com/arextest/storage/service/AgentWorkingService.java [hunk 9]
```diff
@@ -144,6 +147,26 @@ public <T extends Mocker> byte[] queryMockResult(@NotNull T recordItem,
       LOGGER.info("skip main entry mock response,record id:{},replay id:{}", recordId, replayId);
       return zstdJacksonSerializer.serialize(recordItem);
     }
+
+    // sqlParse
+    if (Objects.equals(MockCategoryType.DATABASE, category)) {
+      List<MockerSaveHandler> handlers = mockerHandlerFactory.getHandlers(category);
```
Comment (coryhh) [inline arex-storage-web-api/src/main/java/com/arextest/storage/service/AgentWorkingService.java:153]: Why not directly use "databaseMockerHandel.parseMocker(recordItem)"

File: arex-storage-web-api/src/main/java/com/arextest/storage/service/mockerhandlers/DatabaseMockerHandel.java [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+
+/**
+ * @author niyan
+ * @date 2024/4/23
+ * @since 1.0.0
+ */
+@Component
+@Slf4j
+@AllArgsConstructor
+public class DatabaseMockerHandel implements MockerSaveHandler {
```
Comment (binyu1005) [inline arex-storage-web-api/src/main/java/com/arextest/storage/service/mockerhandlers/DatabaseMockerHandel.java:27]: DatabaseMockerHandel word spelling error

File: arex-storage-web-api/src/main/java/com/arextest/storage/mock/internal/matchkey/impl/DatabaseMatchKeyBuilderImpl.java [hunk 11]
```diff
@@ -168,7 +170,10 @@ private List<byte[]> dbMockKeyBuild(Mocker databaseMocker) {
     byte[] dbNameBytes = CacheKeyUtils.toUtf8Bytes(dbName);
```
Comment (pangdayuan1) [inline arex-storage-web-api/src/main/java/com/arextest/storage/mock/internal/matchkey/impl/DatabaseMatchKeyBuilderImpl.java:170]: At this time, opName has been spliced with dbname and body. Why can't it be obtained directly through operationname?

File: arex-storage-web-api/src/main/java/com/arextest/storage/utils/DatabaseUtils.java [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+    }
+
+    /**
+     * @param operationName example: dbName-tableName, tableName, tableName-action-operationName;
+     * @return tableNames
+     */
+    public static Set<String> parseTableNames(String operationName) {
+        Set<String> tableSet = new HashSet<>();
+        if (StringUtils.isEmpty(operationName)) {
+            return tableSet;
```
Comment (pangdayuan1) [inline arex-storage-web-api/src/main/java/com/arextest/storage/utils/DatabaseUtils.java:79]: There is no need to create a tableSet first when it is empty, you can use Collections.emptySet()

**Review by mr3 — APPROVED** [review]

**Review by pangdayuan1 — APPROVED** [review]

**Pull request merged**